test/reader_test.go

```diff
@@ -0,0 +1,556 @@
+				m.Database("jaeger-tracing").Collection("traces10").Drop(ctx)
+				reader := jaeger_mongodb.NewSpanReader(m.Database("jaeger-tracing").Collection("traces10"), nil)
+				writer := jaeger_mongodb.NewSpanWriter(m.Database("jaeger-tracing").Collection("traces10"), nil)
+				GenerateTraces(ctx, writer, 50, "circular", false)
+				traces200, err := reader.FindTraces(ctx, &spanstore.TraceQueryParameters{
+					StartTimeMin: time.Date(1997, 04, 30, 05, 1, 1, 1, time.UTC),
+					StartTimeMax: time.Now(),
+					NumTraces:    1500,
+					Tags: map[string]string{
+						"http.status_code": "200",
+					},
+				})
+				if err != nil {
+					log.Fatal(err)
+				}
+				// Assert there are 45 traces with tag -- http.status_code = 200
+				assert.Equal(t, 45, len(traces200))
+				traces404, err := reader.FindTraces(ctx, &spanstore.TraceQueryParameters{
+					StartTimeMin: time.Date(1997, 04, 30, 05, 1, 1, 1, time.UTC),
+					StartTimeMax: time.Now(),
+					NumTraces:    1500,
+					Tags: map[string]string{
+						"http.status_code": "404",
+					},
+				})
+				if err != nil {
+					log.Fatal(err)
+				}
+				// Assert there are only 5 traces with tag -- http.status_code = 404
+				assert.Equal(t, 5, len(traces404))
+			},
+		},
+	}
+	for _, tc := range testCases {
+		println(tc.name)
+		tc.runAssertion(tc.endTs, tc.lookback)
+		println("====")
+	}
+}
+func BenchmarkTagFiltering(b *testing.B) {
+	m, err := mongo.Connect(context.TODO(), options.Client().
+		ApplyURI(fmt.Sprintf("mongodb://%s:27017", GetLocalIP())).
+		SetWriteConcern(writeconcern.New(writeconcern.W(1))))
+	if err != nil {
+		log.Fatal(err)
+	}
+	reader := jaeger_mongodb.NewSpanReader(m.Database("jaeger-tracing").Collection("traces"), nil)
+
+	ctx := context.TODO()
+	fmt.Println(reader.GetOperations(ctx, spanstore.OperationQueryParameters{}))
+	fmt.Println(time.Unix(1623086597, 0))
+	fmt.Println("----")
+
+	testCases := []struct {
+		name         string
+		tags         map[string]string
+		runAssertion func(map[string]string)
+	}{
+		{
+			name: "Benchmark test for tag filtering with status code 200",
+			tags: map[string]string{
+				"http.status_code": "200",
+			},
+			runAssertion: func(tags_in map[string]string) {
+				tracesOutput, err := reader.FindTraces(ctx, &spanstore.TraceQueryParameters{
+					StartTimeMin: time.Date(1990, 4, 30, 1, 30, 0, 0, time.UTC),
+					StartTimeMax: time.Now(),
+					Tags:         tags_in,
+					ServiceName:  "customer",
+					NumTraces:    100,
+				})
+				if err != nil {
+					log.Fatal(err)
+				}
+				for _, trace := range tracesOutput {
+					for _, s := range trace.Spans {
+						for _, tag := range s.Tags {
+							if tag.Key == "http.status_code" {
+								assert.Equal(b, int64(200), tag.VInt64, "Status code should be 200")
+							}
+						}
+					}
+				}
+			},
+		},
+	}
+	for _, tc := range testCases {
+		println(tc.name)
+		tc.runAssertion(tc.tags)
+		println("====")
+	}
+}
+
+func TestTagFiltering(t *testing.T) {
```

Comment:
How is this different than `"Test Find Traces -- Tag Filtering",`?

Reply:
This was a test that I wrote a while ago (it was reading off from the existing traces so idempotency was not guaranteed). I have removed this test for now, and will change the benchmark test to use the GenerateTraces function.